lib/ReactViews/Custom/Chart/BottomDockChart.tsx

```diff
@@ -0,0 +1,367 @@
+                />
+              ))}
+              {yAxes.map((y) => (
+                <GridRows
+                  key={`grid-${y.units}`}
+                  width={plotWidth}
+                  height={plotHeight}
+                  scale={y.scale}
+                  numTicks={Y_AXIS_NUM_TICKS}
+                  stroke={yAxes.length > 1 ? y.color : DEFAULT_GRID_COLOR}
+                  lineStyle={{ opacity: 0.3 }}
+                />
+              ))}
+              <svg
+                id="zoomSurface"
+                clipPath="url(#plotClip)"
+                pointerEvents="all"
+              >
+                <rect
+                  width={plotWidth}
+                  height={plotHeight}
+                  fill="transparent"
+                />
+                {cursorX && <Cursor x={cursorX} stroke={DEFAULT_GRID_COLOR} />}
+                <Plot
+                  chartItems={processedChartItems}
+                  initialScales={initialScales}
+                  zoomedScales={zoomedScales}
+                />
+              </svg>
+            </Group>
+          </svg>
+          <Tooltip {...tooltip} />
+          <PointsOnMap chartItems={processedChartItems} />
+        </div>
+      </ZoomX>
+    );
+  }
+);
+
+Chart.displayName = "Chart";
+
+// Type guard to filter ChartItems that don't produce a nearestPoint
+const pointNotUndefined = (itemPoint: {
+  chartItem: ChartItem;
+  point?: ChartPoint;
+}): itemPoint is { chartItem: ChartItem; point: ChartPoint } =>
+  itemPoint.point !== undefined;
+
+/**
+ * Sorts chartItems so that `momentPoints` are rendered on top then
+ * `momentLines` and then any other types.
+ * @param {ChartItem[]} chartItems array of chartItems to sort
+ */
+const sortChartItemsByType = (chartItems: readonly ChartItem[]) => {
+  return chartItems.slice().sort((a, b) => {
+    if (a.type === "momentPoints") return 1;
+    else if (b.type === "momentPoints") return -1;
+    else if (a.type === "momentLines") return 1;
+    else if (b.type === "momentLines") return -1;
+    return 0;
+  });
+};
+
+/**
+ * Calculates a combined domain of all chartItems.
+ * Convert Dates to numbers
+ */
+const calculateDomainX = (chartItems: ChartItem[]) => {
+  const xmin = Math.min(...chartItems.map((c) => +c.domain.x[0]));
+  const xmax = Math.max(...chartItems.map((c) => +c.domain.x[1]));
+  return [xmin, xmax];
+};
+
+const calculateDomainY = (chartItems: ChartItem[]) => {
+  const ymin = Math.min(...chartItems.map((c) => c.domain.y[0]));
+  const ymax = Math.max(...chartItems.map((c) => c.domain.y[1]));
+  return [ymin, ymax];
+};
+
+const findNearestPoint = (
+  points: readonly ChartPoint[],
+  coords: ChartPoint,
+  xScale: XScale,
+  maxDistancePx: number
+) => {
+  function distance(coords: ChartPoint, point: ChartPoint) {
+    // Works with numbers or Dates
+    return point ? +coords.x - +xScale(point.x) : Infinity;
+  }
+
+  let left = 0;
+  let right = points.length;
+  let mid = 0;
+  for (;;) {
```

Comment:
`while (left !== right) {` and remove the break on the first line inside the loop?